### ISSUE TITLE
RGB Picker Tool side effect of zoom/skew

### ISSUE THREAD

**Issue by beeheemooth** [opentoonz/opentoonz#1717]
At first I thought that this thing happens only in OTX, but it turned out that in the Morevna Edition the same bug.
Try to play this bug in OT 1.2 (I have not installed it):
1.Use some non-standard vector fill.
2. Switch to the RGB Piker Tool.
3. For some reason now, this tool has the ability to zoom, which can deform even the camera.
![ezgif-3-a9859e5d2e](https://user-images.githubusercontent.com/25724133/34609912-51f8a206-f227-11e7-8331-22803ade93d8.gif)

**Comment by artisteacher** [comment]
This also happens in 1.2. For this to occur, both of the gradient colors have to be changed from the default. After this occurs, using reset view and fit to window do not fix the extreme zoom or angle distortion for the viewer/camera. Resetting the center of the camera in the schematic does fix the zoom - but not the angle distortion.

After changing to other vector styles & back to the gradient, I can’t seem to trigger the effect. This may not be related but at the same time that the zoom effect stopped, the gradient fill started to display properly in the viewer (in OS X several of the vector styles overflow #891) though they continue to display with the overflow in preview/render and in the drawing icons.

**Comment by RodneyBaker** [comment]
I accidentally ran into this and thought the ability to skew the camera view was very cool.
Later, I wasn't able to get the behavior to repeat but that is very likely because I was no longer using the Color Pick Tool.

**Comment by beeheemooth** [comment]
After "Delete Lines, Styles and Areas" (in this case with a problem style) the behavior of the tools becomes inadequate even if a new scene is created. Only restarting OpenToonz solves this problem.
![ezgif-2-b5297fcfa0](https://user-images.githubusercontent.com/25724133/37691124-446b688c-2cb8-11e8-877b-b820e8d7fd33.gif)

**Comment by DarrenTAnims** [comment]
I've just stumbled across this too in V1.4. The above all describe the issue well. The only thing I've got to add is that it only messes up the viewer in the current room. Change to another room and until you change the colour, the viewer works fine.

**Comment by ladlon** [comment]
Hi.  I was directed to this thread when I posted my own bug report today about this same issue.

I had a scene where I had a character (multi-level, parented), and a photographic background image.
Selecting a part of my character (ex. eyebrow), then selecting the RGB Picker tool, and then clicking on another item in a different level (ex. hair) resulted in what appeared to be (visually) the background image level getting pushed way down the Z axis, resulting in a parallax effect, where the (when the work area is dragged) the character moves with the 'stage' fine, but the background moves very little (and is much smaller than it should be), as if it were far int he distance.

Oddly enough, looking at the 3D view and the Z values in the upper toolbar, it doesn't appear that the background image is on a different Z level than the character, so it's a visual/display effect only (...or the 3D view and Z depth value field are bugged).

What I also noticed, is that when you use the selection tool, and drag out a selection box, the box acts like it's on a different Z plane than the mouse pointer, as it's drastically offset (SouthEast of the mouse pointer), and the scale of the box's stretching as you drag are 'amplified' compared to that of the distance the mouse pointer is moving).

I also noted that the Rectangle mode of the RGB Picker seems to work fine, and so far only the Normal mode has this issue (...I haven't tried the other modes).

(Sort of) glad to see that others are having a similar issue.

Win7Pro, 32gigs RAM, Tahoma1.0
